### ISSUE TITLE
Roadmap to Flux 1.0

### ISSUE THREAD

**Issue by DilumAluthge** [FluxML/Flux.jl#961]
What breaking changes need to be made to the Flux public API before Flux version 1.0.0 can be released?

**Comment by MikeInnes** [comment]
I don't think we know yet, though it's good to start thinking about it. At the least we'll want to let the Zygote work shake out a bit, otherwise new API issues like dropout are going to crop up. Most likely we'll also want to follow something like the Julia release process and have a thorough review of all the documented / exported APIs before setting them in stone.

Most of the peripheral APIs haven't changed much lately, so that could be reasonably easy, but there are at least a couple of things we want to change:

* APIs for RNNs (#648), as more advanced things currently require Zygote internals; perhaps this can be non-breaking though since we don't have specific APIs for it yet.
* Optimisers (#637, @dhairyagandhi96); we might want to punt on this for 2.0, depending on how quickly a design comes together.
